tty.c

```diff
@@ -2258,6 +2266,13 @@ tty_colours(struct tty *tty, const struct grid_cell *gc)
 					tty_putcode1(tty, TTYC_SETAB, 0);
 				tc->bg = gc->bg;
 			}
+			if (COLOUR_DEFAULT(gc->us) && !COLOUR_DEFAULT(tc->us)) {
+				if (have_ax)
+					tty_puts(tty, "\033[59m");
+				else if (tc->us != 0) // TODO: kmo I have no idea what to use here.
```

Comment:
I am a slight bit lost here. My rough understanding is, that because we do not have a way to specify a default underline color, we have to reset everything and then re-apply everything apart from the underline color.

But I do not think I understand the code here well enough to know how to achieve that.

Reply:
Yes you will need to call tty_reset because there is no way to clear the underline colour. That already happens in tty_colours:
~~~~
if (!have_ax && tty_term_has(tty->term, TTYC_OP))
                        tty_reset(tty);